app/uk/gov/hmrc/exports/repositories/PurgeSubmissionsTransactionalOps.scala

```diff
@@ -0,0 +1,103 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.exports.repositories
+
+import org.bson.conversions.Bson
+import org.mongodb.scala.ClientSession
+import org.mongodb.scala.model.Filters
+import play.api.Logging
+import uk.gov.hmrc.exports.config.AppConfig
+import uk.gov.hmrc.exports.models.declaration.ExportsDeclaration
+import uk.gov.hmrc.exports.models.declaration.notifications.{ParsedNotification, UnparsedNotification}
+import uk.gov.hmrc.exports.models.declaration.submissions.Submission
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.transaction.{TransactionConfiguration, Transactions}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class PurgeSubmissionsTransactionalOps @Inject() (
+  val mongoComponent: MongoComponent,
+  submissionRepository: SubmissionRepository,
+  declarationRepository: DeclarationRepository,
+  notificationRepository: ParsedNotificationRepository,
+  unparsedNotificationRespository: UnparsedNotificationWorkItemRepository,
+  appConfig: AppConfig
+)(implicit ec: ExecutionContext)
+    extends Transactions with Logging {
+
+  protected implicit val tc = TransactionConfiguration.strict
+
+  protected lazy val nonTransactionalSession = mongoComponent.client.startSession().toFuture
+
+  def removeSubmissionAndNotifications(
+    submissions: Seq[Submission],
+    declarations: Seq[ExportsDeclaration],
+    parsedNotification: Seq[ParsedNotification],
+    unparsedNotification: Seq[UnparsedNotification]
+  ): Future[Seq[Long]] =
+    if (appConfig.useTransactionalDBOps)
+      withSessionAndTransaction[Seq[Long]](startRemoveOp(_, submissions, declarations, parsedNotification, unparsedNotification)).recover {
+        case e: Exception =>
+          logger.warn(s"There was an error while writing to the DB => ${e.getMessage}", e)
+          Seq.empty
+      }
+    else nonTransactionalSession.flatMap(startRemoveOp(_, submissions, declarations, parsedNotification, unparsedNotification))
+
+  private def startRemoveOp(
+    session: ClientSession,
+    submissions: Seq[Submission],
+    declarations: Seq[ExportsDeclaration],
+    parsedNotification: Seq[ParsedNotification],
+    unparsedNotification: Seq[UnparsedNotification]
+  ): Future[Seq[Long]] =
+    for {
+      submissionsRemoved <- removeSubmissions(submissions, session)
+      declarationsRemoved <- removeDeclarations(declarations: Seq[ExportsDeclaration], session)
+      notificationsRemoved <- removeParsedNotifications(parsedNotification, session)
+      unparsedNotificationRemoved <- removeUnparsedNotifications(unparsedNotification, session)
+    } yield Seq(submissionsRemoved, declarationsRemoved, notificationsRemoved, unparsedNotificationRemoved)
```

Comment:
More for my benefit, but does this for comprehension delete the unparsed first and work backwards?

Reply:
In theory it would flatmap/map from top to bottom, being a transaction would mean all done "in one", however that being turned off would mean submission first. Will refactor to reverse the order.

Reply:
Thanks Tom, I see what you mean now about whether the futures will run serially

Appreciate the research into the answer, looks like the for comprehension can be relied upon (https://stackoverflow.com/questions/24539416/what-is-the-execution-order-sequential-or-concurrent-of-futures-inside-of-for)